lib/active_admin/router.rb

```diff
@@ -19,9 +19,17 @@ def define_root_routes
         if namespace.root?
           router.root namespace.root_to_options.merge(to: namespace.root_to)
         else
-          router.namespace namespace.name, namespace.route_options.dup do
-            router.root namespace.root_to_options.merge(to: namespace.root_to, as: :root)
+          if namespace.name.is_a?(Array)
+            proc = Proc.new do
+              router.root namespace.root_to_options.merge(to: namespace.root_to, as: :root)
+            end
+            (namespace.name.reverse.inject(proc) { |n, c| Proc.new { router.namespace c, namespace.route_options.dup, &n } }).call
+          else
```

Comment:
Could this be simplified by extracting into a method that gets called recursively?

Reply:
I pushed some changes based on what you said, and I do the same in define_namespace.